src/ocrd/processor/base.py

```diff
@@ -200,10 +201,11 @@ def verify(self):
         assert self.output_file_grp is not None
         input_file_grps = self.input_file_grp.split(',')
         output_file_grps = self.output_file_grp.split(',')
-        def assert_file_grp_cardinality(grps, spec, msg):
+        def assert_file_grp_cardinality(grps : List[str], spec : Union[int, List[int]], msg):
             if isinstance(spec, int) and spec > 0:
                 assert len(grps) == spec, msg % (len(grps), str(spec))
             else:
+                assert isinstance(spec, list)
```

Comment:
We are breaking one useful case here: `-1` (unlimited, no checking at all).

```suggestion
            if isinstance(spec, int):
                if spec > 0:
                    assert len(grps) == spec, msg % (len(grps), str(spec))
            else:
                assert isinstance(spec, list)
```

Reply:
Right, my bad, didn't read the if-clause right.